src/Contexts/CurrentProblemState.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+
+type CurrentProblemStateProps = {
+    lastSavedAt?: moment.Moment|null;
+    lastSubmittedAt?: moment.Moment|null;
+    setLastSavedAt?: (val: moment.Moment|null) => void;
+    setLastSubmittedAt?: (val: moment.Moment|null) => void;
+};
+
+const ProblemStateContext = React.createContext<Partial<CurrentProblemStateProps>>({});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const ProblemStateProvider = ({ children }: Props) => {
+    const [lastSavedAt, setLastSavedAt] = useState<moment.Moment|null>(null);
+    const [lastSubmittedAt, setLastSubmittedAt] = useState<moment.Moment|null>(null);
+
+    return (
+        <ProblemStateContext.Provider value={{ 
+            lastSavedAt, 
+            lastSubmittedAt, 
+            setLastSavedAt, 
+            setLastSubmittedAt 
+        }}>
+            {children}
+        </ProblemStateContext.Provider>
+    );
+};
+
+export const useCurrentProblemState = () => React.useContext(ProblemStateContext);
```

Comment:
@doombot-exe I don't think I like the idea of using context here, it seems like a lot of overhead (from a complexity standpoint and probably an efficiency standpoint however that is probably negligible) for passing the state up one level and down to a different level

@PanopticaRising do you have any thoughts on this?